src/PowerShellEditorServices/Utility/PathUtils.cs

```diff
@@ -38,48 +38,22 @@ public static string NormalizePathSeparators(string path)
             return string.IsNullOrWhiteSpace(path) ? path : path.Replace(AlternatePathSeparator, DefaultPathSeparator);
         }
 
-        public static string WildcardEscape(string path)
-        {
-            return WildcardPattern.Escape(path);
-        }
-
         /// <summary>
         /// Return the given path with all PowerShell globbing characters escaped,
         /// plus optionally the whitespace.
         /// </summary>
         /// <param name="path">The path to process.</param>
         /// <param name="escapeSpaces">Specify True to escape spaces in the path, otherwise False.</param>
-        /// <returns>The path with [ and ] escaped.</returns>
+        /// <returns>The path with *, ?, [, and ] escaped, including spaces if required</returns>
         internal static string WildcardEscapePath(string path, bool escapeSpaces = false)
         {
-            var sb = new StringBuilder();
-            for (int i = 0; i < path.Length; i++)
-            {
-                char curr = path[i];
-                switch (curr)
-                {
-                    // Escape '[', ']', '?' and '*' with '`'
-                    case '[':
-                    case ']':
-                    case '*':
-                    case '?':
-                    case '`':
-                        sb.Append('`').Append(curr);
-                        break;
+            var wildcardEscapedPath = WildcardPattern.Escape(path);
```

Comment:
May need to manually handle <kbd>`</kbd> still due to PowerShell/PowerShell#16306.  There is a good chance that this seemingly over engineered logic was here due to issue with that API. Unsure when it was added though.

Reply:
Probably do need to escape it to handle a situation where it exists in a file name at least at the command level, will probably leave it alone at an argument level

Reply:
https://github.com/PowerShell/PowerShellEditorServices/issues/123 is the relevant reasoning, I don't think we need to escape *every* path that comes in, this should be more handled on the output side if required as part of a cmdlet argument, if it's thought the cmdlet will interpret handling characters, because in some cases that may be exactly what the caller wants.

Reply:
I'll try to add a test for SetLineBreakpoint which is the only other place this is referenced.

Reply:
> Probably do need to escape it to handle a situation where it exists in a file name at least at the command level, will probably leave it alone at an argument level

Yeah it gets tricky with no real way to please everyone here.  On one hand you want to enable VSCode variables like `${workspaceFolder}` in which case you can't really escape manually, and on the other hand some folks may want to use expressions.

Like you said though for the command path always escape is a clear winner for sure.

Reply:
@SeeminglyScience the compromise for my implementation is that:
1. Command is always wrapped in double quotes because it is used by the call operator, so ${} expressions will still work if you want to pass a file and not a command, the only file that won't work in this case is if the filename contains $(), I think that's reasonable enough to just say "dont name your files this way", should probably have an explicit exception check for this.
2. Argument is only wrapped in quotes if you specify a space. If you really need to be literal or use workspacefolder in an arg and have it be safe, you should wrap it in quotes, a scriptblock, or $(). I think that's a good compromise

Reply:
@SeeminglyScience looks like all the debugserver tests were commented out by @rjmholt after his commit and probably need to be rewritten. @andschwa can you come up with a way to get a debugservice instantiated to a point I can run tests against it? Been trying to figure it out and it's too much for my non-C# brain at the moment.

Reply:
> Yeah it gets tricky with no real way to please everyone here. On one hand you want to enable VSCode variables like ${workspaceFolder} in which case you can't really escape manually, and on the other hand some folks may want to use expressions.

Wouldn't this be interpolated _before_ it gets to the LSP server? I could be wrong, I just assumed that the Code client would interpolate their own variable expressions, and then send the full string over.

Reply:
> > Yeah it gets tricky with no real way to please everyone here. On one hand you want to enable VSCode variables like ${workspaceFolder} in which case you can't really escape manually, and on the other hand some folks may want to use expressions.
> 
> Wouldn't this be interpolated _before_ it gets to the LSP server? I could be wrong, I just assumed that the Code client would interpolate their own variable expressions, and then send the full string over.

They do, the problem is that you don't know that they are a path or an interpolation once they arrive to the LSP. I think this implemention is a good compromise.